bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieImpl.java

```diff
@@ -618,7 +618,7 @@ private void replay(Journal journal, JournalScanner scanner) throws IOException
                 logPosition = markedLog.getLogFileOffset();
             }
             LOG.info("Replaying journal {} from position {}", id, logPosition);
-            long scanOffset = journal.scanJournal(id, logPosition, scanner);
+            long scanOffset = journal.scanJournal(id, logPosition, scanner, this.conf.isSkipReplayJournalInvalidRecord());
```

Comment:
I have a question:
How do we decide to open the switch
What I understand is that open this switch ,maube cause to loss data,
does this need to discuss it on the dev@ mailing list @eolivelli , have a look,thanks

Reply:
what we meet is 

a bk shutdown because of VM crash,then VM recovered , but we can not restart bk ， because of the exception。we can not skip this， we only can do format data for this VM.. and re install bk


so if we want to recovery the bk in the scene， we can open the switch only on the machine，it will startup,  and next time， we will close this switch